src/components/Correlation/index.js

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import { uuid } from "../../utils";
+
+const createCorrelation = () => {
+  return {
+    lifecycle: {
+      onBeforeEvent(event, options) {
+        return Promise.resolve(options.correlationID).then(correlationID => {
+          if (correlationID !== undefined) {
+            event.correlationID = correlationID;
```

Comment:
ESLint will throw an error on this line without the `no-param-reassign` adjustment I've added to `.eslint.rc`. I feel like assigning a property to event will be something we'll be doing regularly, so I decided to adjust it in `.eslint.rc`. Let me know if you disagree with this.